src/videojs-contrib-hls.js

```diff
@@ -674,12 +674,23 @@ const HlsSourceHandler = function(mode) {
 
       let settings = videojs.mergeOptions(options, {hls: {mode}});
 
+      let previousBeforeRequest;
+
+      if (tech.hls && tech.hls.xhr && tech.hls.xhr.beforeRequest) {
```

Comment:
You don't need to check for `tech.hls.xhr.beforeRequest` here. Even if `beforeRequest` is `undefined`, the next line won't cause an  exception.